inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp

```diff
@@ -18,43 +18,77 @@
 
 using TransformationTests = CommonTestUtils::TestsCommon;
 
-std::pair<bool, std::string> compare_functions(
+class FunctionsComparator {
+public:
+    enum CmpValues {
+        NONE = 0,
+        CONST_VALUES = 1 << 0,
+        NAMES = 1 << 1,
+        RUNTIME_KEYS = 1 << 2,
+        PRECISIONS = 1 << 3,
+        ATTRIBUTES = 1 << 4,
+    };
+
+    struct Result {
+        bool valid;
+        std::string message;
+    };
+
+    static constexpr FunctionsComparator no_default() noexcept {
+        return FunctionsComparator{NONE};
+    }
+    static constexpr FunctionsComparator with_default() noexcept {
+        return FunctionsComparator{PRECISIONS};
+    }
+    FunctionsComparator& enable(CmpValues f) noexcept {
+        m_comparition_flags = static_cast<CmpValues>(m_comparition_flags | f);
+        return *this;
+    }
+    constexpr bool should_compare(CmpValues f) const noexcept {
+        return m_comparition_flags & f;
+    }
+    Result compare(
+        const std::shared_ptr<ngraph::Function>& f1,
+        const std::shared_ptr<ngraph::Function>& f2) const;
+
+    Result operator()(
+        const std::shared_ptr<ngraph::Function>& f1,
+        const std::shared_ptr<ngraph::Function>& f2) const {
+        return compare(f1, f2);
+    }
+
+private:
+    constexpr explicit FunctionsComparator(CmpValues f) noexcept : m_comparition_flags(f) {}
+    CmpValues m_comparition_flags;
+};
+
+///
+/// \deprecated
+/// \brief compare_functions is obsolete function use FunctionComparator instead.
+///
+inline std::pair<bool, std::string> compare_functions(
```

Comment:
Why cannot we remove this method if we implemented the new approach?

Reply:
I assume it should be done in the next step.

Reply:
Ok, let's plan and remove this functionality.